src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -211,6 +214,11 @@ private CommandResult executeCommand(String commandText) throws CommandException
             if (commandResult.isExit()) {
                 handleExit();
             }
+            if (commandResult.isSelect()) {
+                bodyPanel.getAddressPanel().getPersonListPanel().clearSelection();
+                this.setPersonDetail();
+                this.setDisplayedIndex();
+            }
```

Comment:
Referring to the subsequent comment, since you know the selection changes when the listener is triggered, this part can be moved into the listener and `CommandResult` may not need to keep track of whether the command is for selecting.